lib/ui/src/app.tsx

```diff
@@ -47,11 +47,16 @@ const App = React.memo<AppProps>(
           {
             key: 'settings',
             render: () => <SettingsPages />,
-            route: (({ children }) => (
-              <Route path="/settings" startsWith>
-                {children}
-              </Route>
-            )) as FunctionComponent,
+            route: (({ children }) => {
+              if (viewMode === 'settings') {
```

Comment:
viewMode changes over time, but this will never re-evaluate because of useMemo without dependencies..

This will mean the setting page will never, ever render anything...

Reply:
Ah, okay. My apologies, I must have only tested it by loading the settings page by going directly to /settings/about rather than navigating there.

I will go back and take a look at the code base and try and wrap my head around a different solution. I appreciate your patience, as it's my first OSS contribution.

Reply:
>  as it's my first OSS contribution.

Really really appreciated!

> Would simply adding viewMode as a dependency cause unwanted performance issues?

OI'm pretty sure the router gives it as an argument too!